### PR TITLE
Connect demo

### PR DESCRIPTION
**Pull request by rubelux**

### CODE REVIEW

File: src/components/Layout/Modals/LoginModal/lib/index.ts [hunk 1]
```diff
@@ -48,23 +50,60 @@ const useWalletConnect = () => {
...
+    setConnecting(true)
+    const chainId = assertInt(process.env.NEXT_PUBLIC_CHAIN_ID, 'NEXT_PUBLIC_CHAIN_ID')
+
+    try {
+      const { account: publicAddress, chain } = await connectAsync({
+        connector: new WalletConnectConnector({
+          chains,
+          options: {
+            qrcode: true,
+            projectId: '1cfa6214f74719cb6dccea797e0ff417',
```
Comment (EricWVGG) [inline src/components/Layout/Modals/LoginModal/lib/index.ts:67]: should this be… a constant? env var? this is a weird thing to be hard-coded

File: src/components/Layout/Modals/LoginModal/lib/index.ts [hunk 2]
```diff
@@ -48,23 +50,60 @@ const useWalletConnect = () => {
...
+      })
+
+      const challenge = await requestChallengeAsync({ address: publicAddress, chainId: chain.id })
+      if (!challenge) {
+        throw new Error('failed walletconnect challenge')
+      }
+      const { message } = challenge
+
+      const signature = await signMessageAsync({ message })
+      await signIn('moralis-auth', { message, signature, redirect: false, callbackUrl: '/user' })
```
Comment (EricWVGG) [inline src/components/Layout/Modals/LoginModal/lib/index.ts:80]: we don't have a `/user` endpoint anymore, dunno if that matters.

File: src/components/Layout/Modals/LoginModal/lib/index.ts [hunk 3]
```diff
@@ -48,23 +50,60 @@ const useWalletConnect = () => {
...
+
+      setConnecting(false)
+      // Force reload due to JWT is not available or is still linked to old session when first created. Wagmi renders an error when the smart contracts are called.
+      router.reload()
+    } catch (e) {
+      setConnecting(false)
+      console.error('error connecting user', e)
+    }
   }
+  // const connectMetamask = () => {
```
Comment (EricWVGG) [inline src/components/Layout/Modals/LoginModal/lib/index.ts:90]: if we're not going to delete this, we should at least explain why it's commented out and shouldn't be used.

**Review by EricWVGG — APPROVED** [review]

File: src/components/Layout/Modals/LoginModal/lib/index.ts (hunk 2 above)
Comment (rubelux) [reply src/components/Layout/Modals/LoginModal/lib/index.ts:80]: it's been there for ages, I believe it does not matter, but I will review it

File: src/components/Layout/Modals/LoginModal/lib/index.ts (hunk 1 above)
Comment (rubelux) [reply src/components/Layout/Modals/LoginModal/lib/index.ts:67]: Totally, let's check if it works and I put it into a const

**Pull request merged**